### ISSUE TITLE
Keyboard accessibility

### ISSUE THREAD

**Issue by adammockor** [arendjr/selectivity#7]
Hello,

would by nice to have better keyboard accessibility. For example when you pick option in dropdown (click or enter), select lose focus, so you cannot opened it again with enter.

Select with submenu is not accessible at all with keyboard.

Otherwise nice modularised design.

**Comment by arendjr** [comment]
Thanks, and good points. I'll try to pick this up soon!

**Comment by msx2** [comment]
@arendjr Speaking of keyboard accessibility... there's another issue that should be resolved.

Typically when you use keyboard only to go through the page, you use keys like `tab` or `shift`+`tab` to go to the next or previous elements on the page. But since not all of the selectivity containers are rendered as inputs/links/whatever that is accessible, and just plain `div`s/`span`s, you don't have any way of accessing those elements.

**Comment by arendjr** [comment]
Okay, first things first :) I've just committed a6ea06b. This adds a hidden (opacity: 0) input to all single-value selectors which adds them to the tab navigation. Also, it enables keyboard support for single-value selectors without search input.

Next on the list: Re-opening on Enter and then improved keyboard accessibility for submenus.

**Comment by arendjr** [comment]
Good, keeping the focus so you can reopen it with Enter (and other keys) wasn't too hard: 8c1526 and 445d13.

I think accessibility for submenus will be a bit harder though :)

**Comment by arendjr** [comment]
Made some progress towards accessibility for submenus with b1a1d15. Moving up and down now moves through the deepest nested submenu by default. Using backspace you can close this submenu, allowing you to navigate at higher levels. The most awkward remaining issue is that if you want to move multiple positions at a high level, but submenus open for intermediate results, you have to manually close all those along the way.

**Comment by dlong500** [comment]
It would also be good to have keyboard support for deleting a value on fields where `allowClear` is true.

**Comment by arendjr** [comment]
Alright, I've added support for clearing the select box on backspace if `allowClear` is true and the dropdown is closed.

Navigation through submenus has been improved as well.

Hopefully all keyboard related stuff has been fixed for now, but if everyone interested wants to test things out that would be very welcome. If all is well, I hope to create a new release within the week.

**Comment by mgifford** [comment]
I do think this should be re-opened as I think there are some outstanding issues when looking at https://arendjr.github.io/selectivity/

I wasn't able to effectively navigate through the examples without a mouse.
